File: src/main/java/com/dyrosoft/kvbparser/parser/DepartureInformationParserFunc.java
package com.dyrosoft.kvbparser.parser;

import com.dyrosoft.kvbparser.utils.StringUtils;
import com.google.common.collect.ImmutableList;

import org.jsoup.Jsoup;
import org.jsoup.nodes.Document;
import org.jsoup.nodes.Element;
import org.jsoup.select.Elements;

public class DepartureInformationParserFunc
        extends AbstractRxHtmlParserFunc<ImmutableList<String>> {

    @Override
    protected ImmutableList<String> parse(final String html) {
        final ImmutableList.Builder<String> builder = ImmutableList.builder();

        final Document document = Jsoup.parse(html);
        final Elements elements = document.select("table.qr_table");
        for (final Element tr : elements.get(0).getElementsByTag("tr")) {
            final Elements tds = tr.getElementsByTag("td");
            builder.add(StringUtils.advancedTrim(tds.get(0).text()));
        }
        return builder.build();
    }
}

File: src/main/java/com/dyrosoft/kvbparser/parser/DepartureInformationParserFunc.kt
package com.dyrosoft.kvbparser.parser

import com.dyrosoft.kvbparser.utils.StringUtils
import org.jsoup.Jsoup

class DepartureInformationParserFunc : AbstractRxHtmlParserFunc<List<String>>() {

    override fun parse(html: String): List<String> {
        val list = mutableListOf<String>()
        val document = Jsoup.parse(html)
        val elements = document.select("table")
        elements[0].getElementsByTag("tr")
                .map { it.getElementsByTag("td") }
                .forEach {
                    StringUtils.advancedTrim(it[0].text())?.run {
                        list.add(this)
                    }
                }
        return list.toList()
    }
}
